acceptance/tests/resource/service/ticket_5024_systemd_enabling_masked_service.rb

```diff
@@ -0,0 +1,47 @@
+test_name 'Systemd masked services are unmasked before attempting to start'
+
+# This test in intended to ensure that a service which was previously marked
+# as masked and then set to enabled will first be unmasked.
+# Confine to RedHat 7, which uses the systemd provider.
+confine :to, :platform => 'el-7'
+confine :to, {}, hosts.select { |host| host['template'] =~ /redhat/ }
+
+manifest_install_httpd = %Q{
+  package { 'httpd':
+    ensure => present,
+  }
+}
+
+manifest_httpd_masked = %Q{
+  service { 'httpd':
+    provider => systemd,
+    enable => mask,
+  }
+}
+
+manifest_httpd_enabled = %Q{
+  service { 'httpd':
+    provider => systemd,
+    enable => true,
+    ensure => running,
+  }
+}
+
+agents.each do |agent|
+  step "Installing httpd"
+  apply_manifest_on(agent, manifest_install_httpd, :catch_failures => true)
+
+  step "Masking the httpd service"
+  apply_manifest_on(agent, manifest_httpd_masked, :catch_failures => true)
+  on(agent, puppet_resource('service', 'httpd')) do
+    assert_match(/ensure => 'stopped'/, stdout, "Expected httpd service to be stopped")
+    assert_match(/enable => 'false'/, stdout, "Expected httpd service to be masked")
```

Comment:
i think the symlink not existing is not enabled but not masked.  i forget some of those details.  should probably read up on what masking entails.  i think it might actually set the init script itself to /dev/null, so trying to start that service fails.  or maybe it's per-runlevel.
to check where a link points use `readlink` if it's on the host, or `ls -l` and parse out the -> entry